alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -374,6 +374,13 @@ class DocumentLevelMonitorRunner : MonitorRunner() {
                 // Clean up any queries created by the dry run monitor
                 monitorCtx.docLevelMonitorQueries!!.deleteDocLevelQueriesOnDryRun(monitorMetadata)
             }
+
+            if (monitor.dataSources.queryIndex.contains("optimized")) {
```

Comment:
plz add a flag in monitor creation called `deleteQueryIndexBeforeEveryRun` (or something briefer)

Reply:
added as part of pr https://github.com/opensearch-project/common-utils/pull/734